bson/src/main/org/bson/BsonDocument.java

```diff
@@ -782,6 +782,17 @@ public String getFirstKey() {
         return keySet().iterator().next();
     }
 
+    /**
+     * Gets the first value in the document
+     *
+     * @return the first value in the document
+     * @throws java.util.NoSuchElementException if the document is empty
+     * @since 3.7
+     */
+    public BsonReader asBsonReader() {
```

Comment:
This allows the command logger to get the most efficient BsonReader for whatever subclass of BsonDocument it has.